src/classes/hotel.js

```diff
@@ -0,0 +1,110 @@
+import Customer from './customer';
+import Room from './room';
+import Booking from './booking';
+import roomsData from '../../test/test-data/room-data';
+
+class Hotel {
+  constructor(roomsData, customersData, bookingsData) {
+    this.roomsData = roomsData;
+    this.customersData = customersData;
+    this.bookingsData = bookingsData;
+    this.allRooms = [];
+    this.allBookings = [];
+    this.allCustomers = [];
+  }
+  
+  addRooms() {
+    this.roomsData.forEach(roomData => {
+      let room = new Room(roomData);
+      this.allRooms.push(room);
+      return room;
+    })
+  }
+
+  addBookings() {
+    this.bookingsData.forEach(bookingData => {
+      let booking = new Booking(bookingData);
+      this.allBookings.push(booking);
+      return booking;
+    })
+  }
+
+  addCustomers() {
+    this.customersData.forEach(customerData => {
+      let customer = new Customer(customerData);
+      this.allCustomers.push(customer);
+      return customer;
+    })
+  }
+
+  filterBookingsByCustomerID(customer) {
+    let filteredBookingsById = this.allBookings.filter(booking => customer.id === booking.userID);
+    return filteredBookingsById
+  }
+
+  sumUserBookings(customer) {
+    let filteredBookingsById = this.allBookings.filter(booking => customer.id === booking.userID);
+    let sum = filteredBookingsById.reduce((sum, booking) => {
+      this.allRooms.forEach(room => {
+        if (room.roomNumber === booking.roomNumber) {
+          sum += room.cost
+        }
+      })
+      return sum
+    }, 0)
+    return sum
+  }
+
+  filterRoomsByDate(dateInput) {
+    let formatInputDate = dateInput.split('-');
+    let properInputDate = formatInputDate.join('/');
+    let filteredRoomsByDate = []
+    this.allRooms.filter(room => {
```

Comment:
This is quite something! And I mean that with the most admiration! Especially since you said it worked THE FIRST TIME. I say keep as is and if it works in your project - awesome, leave it. There are ways to make it shorter, but I don't personally have the know-how with how you're approaching it to give any guidance. Sorry about that.